File: 1-introdução/ContaBancaria2/src/class/ContaBancaria.ts


export class ContaBancaria {
    private saldo: number
    public titular: string

    constructor(titular: string, saldoInicial=0){
        this.titular = titular
        this.saldo = saldoInicial
    }

    //metodo publico para consultar o saldo
    getSaldo(){
        return this.saldo
    }

    // metodo publico para depositar
    depositar(valor: number){
        if(valor > 0){
            this.saldo += valor
            console.log(`Depósito de R$${valor} realizado.`)
        } else {
            throw new Error("Valor de depósito invalido.")
        }
    }

    // metodo publico para sacar
    sacar(valor:number){
        if(valor > 0 && valor <= this.saldo){
            this.saldo -= valor
            console.log(`Saque de R$${valor} Realizado.`)
        } else {
            throw new Error("Saldo insuficiente ou inválido.")
        }
    }


}

File: 3-POO-LOGIC-WITH-PYTHON/01-INTRODUCAO/CLASSE-CONTABANCARIA-V1/classes/Conta_bancaria.py


class Conta_bancaria:
    def __init__(self, saldo:float=0.0):
        self.saldo = float(saldo)
    
    
    def depositar(self, valor:float):
        self.saldo += valor
    
    def getSaldo(self) -> float:
        return self.saldo
    
    def sacar(self, valor:float):
        if(valor > 0 and valor <= self.saldo):
            self.saldo -= valor
        else:
            return "Saldo insuficiente."